### PR TITLE
test: add test for offer parsing with empty paths and valid issuer

### PR DESCRIPTION
**Pull request by erickcestari**
Verifies that offers with empty paths are accepted as long as the `issuer_id` is set.

Follow-up: #4018

### CODE REVIEW

File: lightning/src/offers/offer.rs [hunk 1]
```diff
@@ -2002,6 +2002,13 @@ mod tests {
 			panic!("error parsing offer: {:?}", e);
 		}
 
+		let mut builder = OfferBuilder::new(pubkey(42));
+		builder.offer.paths = Some(vec![]);
```
Comment (TheBlueMatt) [inline lightning/src/offers/offer.rs:2006]: nit: just use `builder.clear_paths()`? If you really want you could even check that the offer built in `pays_for_offer_without_blinded_paths` round-trips.

Comment (erickcestari) [reply lightning/src/offers/offer.rs:2006]: The ideia is to test this specifc scenario:

Where path is `Some(empty_vector)` and with a valid `issuer_id`

since we moved from:

```rust
let (issuer_signing_pubkey, paths) = match (issuer_id, paths) {
			(None, None) => return Err(Bolt12SemanticError::MissingIssuerSigningPubkey),
			(_, Some(paths)) if paths.is_empty() => return Err(Bolt12SemanticError::MissingPaths),
			(issuer_id, paths) => (issuer_id, paths),
		};
```

to:
```rust
let (issuer_signing_pubkey, paths) = match (issuer_id, paths) {
			(None, None) => return Err(Bolt12SemanticError::MissingIssuerSigningPubkey),
			(None, Some(paths)) if paths.is_empty() => {
				return Err(Bolt12SemanticError::MissingPaths)
			},
			(issuer_id, paths) => (issuer_id, paths),
		};
```
so before the update this new test would fail.

`builder.clear_paths()` would insert `None` into `offer.paths` which isn't the case we want to test

```rust
	#[cfg_attr(c_bindings, allow(dead_code))]
	pub(crate) fn clear_paths($($self_mut)* $self: $self_type) -> $return_type {
		$self.offer.paths = None;
		$return_value
	}
```

Comment (TheBlueMatt) [reply lightning/src/offers/offer.rs:2006]: Mm, then we need to test actually *paying* such a blinded path, not just decoding it.
